File: tests/fixtures/ruby_service_app/handlers/user_handler.rb
require_relative '../services/user_service'

def build_service
  UserService.new
end

def handle_create_user(name)
  # Parenthesized deliberately: tree-sitter parses a parenless `build_service`
  # as an identifier, not a call, so the extractor does not see it.
  service = build_service()
  # Variable receiver: deliberately unresolved.
  service.create_user(name)
end

File: tests/fixtures/java_service_app/src/main/java/com/example/handlers/UserHandler.java
package com.example.handlers;

import com.example.services.UserService;

public class UserHandler {

    public String handleCreateUser(String name) {
        // Class-qualified static call: resolves through the name registry.
        return UserService.createUser(name);
    }

    public String handleDescribe() {
        UserService service = new UserService();
        // Variable-qualified call: tree-sitter cannot tell this from a
        // class-qualified one, so it must stay unresolved (D34).
        return service.describe();
    }
}
